style/ruby/README.md

```diff
@@ -411,3 +424,239 @@ hash = { one: 1, two: 2, three: 3 }
+
+```ruby
+# ruim
+# e/ou em condições (sua precedência é baixa, pode produzir resultado inesperado)
+if got_needed_arguments and arguments_valid
+  # ...body omitted
+end
+
+# no cálculo da expressão lógica
+ok = got_needed_arguments and arguments_valid
+
+# bom
+# &&/|| em condicionais
+if got_needed_arguments && arguments_valid
+  # ...body omitted
+end
+# no cálculo da expressão lógica
+ok = got_needed_arguments && arguments_valid
+
+```
+
+## Não deve haver espaço depois do `!`
+
+Sem espaço após `!`.
+
+```ruby
+#ruim
+# Verifica se o array não está vazio
+if ! array.empty?
+  puts "O array tem #{array.size} elementos"
+end
+
+# bom
+# Verifica se o array não está vazio
+if !array.empty?
+  puts "O array tem #{array.size} elementos"
+end
+
+```
+
+## Dupla Negação
+
+Evite o uso de `!!`.
+
+`!!` converte um valor para booleano, mas você não precisa dessa conversão explícita na condição de uma expressão de controle; usá-la só torna sua intenção mais complexa. Se você quer fazer uma verificação de `nil`, use `nil?` em vez disso.
+
+```ruby
+# ruim
+x = 'test'
+# cheque nulo obscuro
+if !!x
+  # body omitted
+end
+
+# bom
+x = 'test'
+if x
+  # body omitted
+end
+```
+
+## Operador ternário (`?:`) vs `if`
+
+Prefira o operador ternário (`?:`) sobre as construções `if/then/else/end`. É mais comum e obviamente mais conciso.
+
+`condição ? expressão_se_verdadeira : expressão_se_falsa`
+
+Aqui está um exemplo de como usar o operador ternário para determinar se um número é par ou ímpar:
+
+```ruby
+# ruim
+if num % 2 == 0
+  "par"
+else
+  "ímpar"
+end
+
+# O mesmo exemplo usando operador ternário:
+
+# bom
+num % 2 == 0 ? "par" : "ímpar"
+
+```
+
+## `case` ao invés de `if-elsif`
+
+Prefira o uso de `case` ao invés de `if-elsif` quando o valor comparado é o mesmo em cada cláusula.
+
+```ruby
+# ruim
+if status == :active
+  perform_action
+elsif status == :inactive || status == :hibernating
+  check_timeout
+else
+  final_action
+end
+
+# bom
+case status
+when :active
+  perform_action
+when :inactive, :hibernating
+  check_timeout
+else
+  final_action
+end
+```
+## Retornando Resultado do `if/case`
+
+Aproveite o fato de que if e case são expressões que retornam um resultado.
+
+```ruby
+# ruim
+if condition
+  result = x
+else
+  result = y
+end
+
+# bom
+result = if condition
+           x
+         else
+           y
+         end
+
+# ruim
+case x
+when 1
+  result = "x is equal to 1"
+when 2
+  result = "x is equal to 2"
+when 3
+  result = "x is equal to 3"
+else
+  result = "x is not 1, 2, or 3"
+end
+
+# bom
+result = case x
+         when 1
+          "x is equal to 1"
+         when 2
+          "x is equal to 2"
+         when 3
+          "x is equal to 3"
+         else
+          "x is not 1, 2, or 3"
+         end
+```
+
+## `unless` em vez de `if`
+
+Preferir `unless` em vez de `if` para condições negativas (ou fluxo de controle `||`).
```

Comment:
>`if` e `unless` são estruturas de controle condicionais em Ruby, que permitem que o código execute determinadas ações apenas se a condição especificada for atendida. A diferença principal entre eles é a direção da condição: `if` executa a ação se a condição for verdadeira, enquanto `unless` executa a ação se a condição for falsa.
O `if` e `unless` são muito úteis em situações em que você precisa verificar uma condição antes de executar uma ação.
Em geral, `if` é a estrutura de controle mais comum em Ruby e é geralmente a mais clara e fácil de entender. No entanto, em algumas situações, `unless` pode ser uma escolha mais clara e legível.
Além disso, `unless` pode ajudar a evitar erros comuns, como esquecer de negar uma condição em uma expressão condicional.
No entanto, é importante lembrar que `unless` deve ser usado com moderação e apenas para condições negativas simples. Para condições mais complexas, é melhor usar `if` com uma condição positiva.
https://rubystyle.guide/#unless-for-negatives